### DISCUSSION TITLE
Issues with Messenger Integration

### DISCUSSION THREAD

**Discussion by sojan-official** [chatwoot/chatwoot#3397] [Self hosted] [answered]
This discussion is to capture the common issues that the community encountered while setting up the Facebook/ Instagram channel

### Facebook not approving the app

ref: https://github.com/chatwoot/docs/issues/114

###  Invalid Scopes: instagram_basic. 

> For Your test app standard access would work, for that, you would need to Add a Product "Instagram Basic Display"
> For the production app, you need advanced access.
> 
> In the dashboard page you should be able to see instagram basic display option
> 

<img width="1439" alt="Screenshot_2021-11-12_at_7 38 30_PM" src="https://user-images.githubusercontent.com/73185/141984915-4002358d-ff32-48bd-8d5a-dc3d29a81ffc.png">

ref: https://discord.com/channels/647412545203994635/647729604944986124/908720660040269854

-----
### JSSDK Option is Not Toggled

Go to your FB developer page, choose your FB app, for Facebook login settings, and turn on the SDK option
![fb_login](https://user-images.githubusercontent.com/73185/144567664-2572d7ee-bef9-452e-bdb1-992ad22fedd4.png)

ref: https://discord.com/channels/647412545203994635/647412545686208527/915722440364163142

-----

###  How to create test apps ? 

<img width="461" alt="Screenshot_2021-11-12_at_7 51 05_PM" src="https://user-images.githubusercontent.com/73185/141985047-885a2c23-dcd9-409b-b803-abf6326e6047.png">

ref: https://discord.com/channels/647412545203994635/647729604944986124/908723268201103370

-----

### Not able to receive Instagram messages, but only the Facebook messages ?

> Hi all i have connected facebook application and linked my Instagram business account as well but i m not able to receive any message . i got message if i do testing from facebook 
>  subscription page. Just stuck and dont know where to move forward or even how to debug the issue.
> surprisingly getting facebook messages but not instagram messages 

ref: https://discord.com/channels/647412545203994635/647729604944986124/909853503416070214

If it's your test app you should add the Instagram testers account to the roles to send and receive the messages. If it's not the test app then your Facebook app should have permission for Instagram messages.
https://developers.facebook.com/docs/instagram-basic-display-api/getting-started/

-----

followed everything and still doesn't work?  add a comment and the team will help you.

**Comment by tejaswinichile** [comment, marked as answer]
Your submission details for `pages_show_list, pages_manage_metadata, pages_messaging, Business Asset User Profile Access`

Login to the account using the following credentials.

Email: facebook_mail_address
Password: Facebook password

1. Click on Settings.
2. Click on Inboxes
3. Click on Facebook
4. Click on login with facebook
5. Authenticate with facebook and select a page.
6. Create inbox.
7. Send a message to the page, you would be able to see it in the dashboard.


Permissions and the answer to facebook's question `Tell us how you're using this permission or feature`

**Business Asset User Profile Access**
> Tell us how you're using this permission or feature

This access is used to get the business name and picture so that it can be used to create inboxes in your-site-name.

**pages_messaging**
> Tell us how you're using this permission or feature

Send message to the users to support them. The access will be used to send support to the users.

> As part of the review process, we will check that the functionality of the app experience is working as intended. If you provide a Page management surface to users, provide us with a temporary test account so we can test it.

Created facebook test account with page access.

> Please provide step-by-step instructions to walk us through how to test this integration.

Connect the page to the account shared before. 
1. Send a message to the page with the account. 
2. The message would be available on the your-site-name website. 
3. Send a reply message from the your-site-name website, it would be available as a message to the end-user.

** pages_manage_metadata**

> Tell us how you're using this permission or feature

Get the details of the page to add enough context for the agents attending the conversations.

**pages_show_list**

> Tell us how you're using this permission or feature

List all pages of a particular person so that they can connect it to receive messages.

```
NOTE: With this, you need to submit the video of the working Facebook app on the Facebook test app.
```

**Reply by bielvelozo** [reply]
And why do chatwoot also use business_management?

**Reply by Braingearceo** [reply]
Hey, did you managed to get advanced access for business_management?

**Comment by tejaswinichile** [comment]
Your submission details for `instagram_manage_messages, instagram_basic `

You can use the following to test the integration.

Email: facebook_mail_address
Password: Facebook password

**instagram_basic**

1. Click on settings -> Inboxes -> Add inbox.
2. Click on Messenger
3. Login with Facebook
4. Select Instagram profiles
5. Select the page Instagram is connected to
6. Click continue and select the page name in the next step.
7. Send a message to Instagram.

It will start showing in the your-app-name dashboard and you can reply from there.

> Tell us how you're using this permission or feature

Users can connect their professional Instagram accounts and chat with their customers. Effectively they would manage all the customer support from the dashboard.

Make video similar to this with your self-hosted version: https://youtu.be/BnAGfjvIUVU

**instagram_manage_messages**

Go to our testing website.

1. Login to the website with the below credentials
Username: **
Password: **
2. Click on: Inboxes
3. Click on: Add Inbox
4. You will see "Choose a channel Page"
5. Click on "Messenger"
6. Click on "Continue with Facebook"
7. Login with Facebook with given credentials:
Username: **
Password: **
8. You will see “Continue As Web App” click on that.

- Click on the "Edit setting" in the same dialogue box.

- In the same dialogue, select on the checkbox to authorize Instagram basic setting for "instagram_account" click on next
Username: **
Password: **
- And authorize "Facebook page" with all the accesses.

9. You will be redirected to the your-app-name webpage again to fill in Inbox Details.
10. Click on: Chose page
11. Select:  "Facebook page" click next.
12. Click on “Create Inbox”
13. Select Agent: from the dropdown click on agent_name
14. Click on “Add Agents”
15. Click on “Take me there”
16. And your Inbox is ready to receive messages which you will receive in your connected Instagram account.


Open a new tab
1. Go to the Instagram login page: https://www.instagram.com/
Login with the below credentials as an Instagram tester
Username: instagram_id
Password: **
2. Go to Instagram DM page: https://www.instagram.com/direct/inbox/
3. Click on instagram_id DM box to send a message, which is connected to our Facebook app.
4. Send some messages to instagram_id account, as shown in the video.

Go to your-app website tab in the browser opened above
1. Switch to your-app website window now.
2. Refresh the page
3. Click on: “Facebook Page”
4. Click on: "Conversation Contact"
5. You will see the message sent from the Instagram account with the Instagram user id name.
6. You can reply to that message and check the Instagram account inbox.
7. Check the Instagram direct inbox

Just to verify more, This is the Instagram account connected to our Facebook app. Now, switch to Instagram account tab.

Switch Instagram account, Credential to Instagram account which is connected to the Facebook app
Username: instagram_id
Password: **

You can check DMs here from fa.cebookqa2 and messages sent from your-app website these are the same messages that we getting to our application through the webhook URLs

You can add more Instagram testers through the Facebook test app to test this feature with more accounts.

> Tell us how you're using this permission or feature

Whenever users send a message to Instagram account by using Facebook webhook URL we get it on our website.

We are a customer support company and some of our clients use Instagram to publicize their products. And they use our customer support app on their website.

So they would like to see all customer queries in one place.
So whenever they get any product-related question on the Instagram account that would get redirected to our site and our client can directly reply from our website to that Instagram user.

For this, we integrated the Instagram webhook, and now to get the queries from the real-time customers on the live app we need your approval on the instagram_manage _massages so that we would fetch the messages from Instagram sent to our Instagram account and display them on our website.

Thanks.


Make video similar to this with your self-hosted version: https://youtu.be/BnAGfjvIUVU

**Reply by nguyendinhlam88** [reply]
Facebook staff not use facebook test account to login facebook for authenticate (they can not login and test features, i can not add their account into whitelist as dev or test), any solution for this problem.

**Reply by agutier6** [reply]
Hi, 

I used your explanation here for the instagram_basic permission, and my app got rejected. Any idea why? This is the response they gave me:

We have determined that your app's use case for the requested permission is invalid or is not needed to support its core functionality.
Your app should only request for permissions that help enhance and build a solid user experience for your business and the end user. Please resolve this issue by removing or revising your app content for compliance before submitting for review.
If you believe your use case is valid, please consider submitting a revised screencast or notes that explains the following items:
1. Which functionality of your app requires the permission.
2. How will the permission to enhance your app's functionality and how the integration functions.
3. How will the permission enhance end user's experience.

I really dont know specifically how chatwoot uses the instagram_basic permission, and that seems to be what they are asking. Could you help me with that please?

Thanks for your help beforehand!

**Reply by pierrecorsini** [reply]
Same issue here; rejected and I'm not even sure why this permission is required actually.

> While reviewing this submission, we were unable to verify the permission because of the use case. Specifically, we noticed that your use case details is not valid for how the permission is supposed to be used.
> 
> Specifically, the use case for this permission is not valid since it does coincide with IG Messaging.
> 
> Instagram Basic grants the app permission to read IG accounts of users that the app manages. This means access to their profile information and media files (example end-points: biography, followers_count, follows_count, media_count, name, profile_picture_url, username, website, media, including media URL, comments under media and permalinks.) This is different from IG Basic Display as this permission is meant for Instagram Business accounts, not consumer.Instagram Basic is the foundation for all Instagram Graph API permissions.
>

**Reply by rafaelstelles** [reply]
Hello,

   In my case, facebook rejected for this problem.

> Hello Team,
> 
> 
> We have determined your use case is valid. While reviewing your submission, we were unable to verify use case in your screencast. Please provide screencast where your use case is reflected. Please show username, bio, profile picture, ID, followers and the following.
> 
> Please re-submit accordingly and we will re-review your application. Thanks!

**Comment by tejaswinichile** [comment]
Make sure you follow this documentation first for the self-hosted version.

https://www.chatwoot.com/docs/self-hosted/configuration/features/integrations/instagram-channel-setup

https://www.chatwoot.com/docs/self-hosted/configuration/features/integrations/facebook-channel-setup

**Comment by tejaswinichile** [comment]
Submission details for `pages_read_engagement`

Pages read engagements permission is used to get Instagram id connected to page so as to have the ability to send messages to customers.

You can use the following to test the integration.

Email: facebook_mail_address
Password: Facebook password

1. Click on settings -> Inboxes -> Add inbox.
2. Click on Messenger Icon[This redirects to Facebook login]
3. Login with Facebook
4. Select Instagram profiles
5. Select the page Instagram is connected to
6. Click continue and select the page name in the next step.
7. Send a message to Instagram.
8. It will start showing in the your-app-name dashboard and you can reply from there.

`pages_read_engagement`

**Tell us how you're using this permission or feature.**

I am the Facebook app administrator I can read the Instagram id connected to the page but when another user is just a page owner he/she is not able to read the instagram_id connected to my page, and that's why other page owners are not able to connect their page with Instagram to our inbox. 

When any user will send a message to the connected Instagram account we want to display the same in our Chatwoot inbox.

We need to check the Instagram id of the account which is connected to our FB page, to set it as the receiver account.
So whenever any user adds a new Facebook inbox to our website we will be able to read the attached Instagram id to the page and we will read the further Instagram message sent to our inbox.

**Reply by patipol** [reply]
I've used this submission details for Facebook App Review Requests but not approved. Review feedback said that "... the explanation of your app's use case is unclear."

Here are the details of review feedback:
<img width="913" alt="Screen Shot 2566-08-12 at 12 09 44" src="https://github.com/chatwoot/chatwoot/assets/580037/be1fd21a-21b1-47f7-8508-3f1d6ed2d0a5">
<img width="914" alt="Screen Shot 2566-08-12 at 12 09 36" src="https://github.com/chatwoot/chatwoot/assets/580037/bd436003-2233-4f27-8905-40671dd98f46">

Could anyone please help me figure this out and get approved?

**Comment by tejaswinichile** [comment]
Please note:

If you are recently facing any issues in receiving messages in the Chatwoot Facebook page inbox, please make sure your page is not connected to any other apps.

According to the new Facebook handover protocol 
https://developers.facebook.com/docs/messenger-platform/handover-protocol/ 
If there are multiple apps connected to the Facebook page, then the primary app in this case `Any other app than the Chatwoot` should release the thread first on the Facebook conversation then we will be able to get the callback on messages received on messenger.

<img width="1358" alt="Other App" src="https://user-images.githubusercontent.com/10827585/164226770-31c07520-aa63-42a7-aecd-0fe0b3c65d88.png">



To resolve this issue:

1. You need to remove the extra app connected to the same facebook other than Chatwoot.
To this solution, you need to remove the other app's connection and make chatwoot a primary app from 
page -> settings -> Advanced Messaging -> Handover protocol -> Messenger receiver
here is the screenshot for the same: 

If you don't remove the other app and just make the Chatwoot a primary app there is no guarantee that we would receive any messages from the users who already sent the messages to the tier5 page from 14th March until now as the thread hasn't been released by the previous app.

2. If you can't remove the first app then we are building one workaround to get the `standby` messages but that would take time to build and test and add it into release from our side, as mentioned here we should use standby messaging if thread control is not implemented: 
https://developers.facebook.com/docs/messenger-platform/handover-protocol/#app-type

For the latter solution, we would need to wait until the next release.

Thank you to the whole team and the feedback, it is that why we have been able to find the root cause for the same.

**Reply by napter** [reply]
I think the UI changed - I don't see advanced messaging. Is there another way to check this?

**Comment by tejaswinichile** [comment]
Human-Agent permission for replying to messages on Instagram after the messaging window is closed.

The following are the steps that must be completed in order for your "Human Agent" permission to be approved:

1. An app can be approved for the Human-Agent tag if they were approved for either instagram_manage_messages or pages_messaging before.

2. This should be demonstrated in the screencast provided by you, i.e. them using pages_messaging or instagram_manage_messaging in your app.

3. App must be able to demonstrate that you have live agent experience to respond to threads.
-A screencast video (sample) showing how the app allows a user to connect their Instagram business account and provide human agent support in their app using the Instagram Messaging API (instagram_manage_messaging or pages_messaging) The screencast should include:
- Scenarios for sending text messages from both an Instagram and Facebook Business Page to a test user.
The App UI shows receipt of that message along with the functionality to send a message as a support agent.

**Comment by grupale** [comment]
¿How can I upgrade my self hosted installation to support Facebook Api v14? Thanks

**Reply by tejaswinichile** [reply]
@grupale  We had raised the PR for the same and it is available in v2.6.0.
[This](https://github.com/chatwoot/chatwoot/commit/374b3671154d65df302130c07037f50438164ffa#diff-8bf37a019cab1dedea458c437bd93e34af1d6e22b1672b1d43ef6eaa4dcb7732R46) is the change for the same:

**Comment by weppa-cloud** [comment]
Hi
We Follow all steps and we have approved Facebook App already. but we requested these two permissions for Instagram but they did not approve them. We can't ask Facebook team because we don't have available any chat or channel support.

[Instragram app revew](https://user-images.githubusercontent.com/69656828/187245014-9f6b9ebb-6e0b-494f-8175-1a729c404f28.png)

Is there something in the new Instagram app that we are missing?

**Reply by tejaswinichile** [reply]
@weppa-cloud Recently all the reviews related to Instagram are getting dispproved.
But if your Instagram app working in the development mode they should have approved it. What reason did they mention while disapproving it?
If you open the rejected review, you will see messenger/facebook bubble on that page where you can talk to the reviewing side of people.

**Reply by Liliadetassia** [reply]
@tejaswinichile tbm estou tendo reprovação só no Instagram

**Comment by ptd-31** [comment]
Hello @tejaswinichile , I have a question. How can i create inbox facebook and connect to facebook account without config file .env(FB_VERIFY_TOKEN, FB_APP_SECRET, FB_APP_ID). Because i logged website: https://app.chatwoot.com/ create channel facebook don't need config file .env and don't need fill information FB app. It's so fast.  I hope to reiceive answer from team. Thanks.

**Reply by sojan-official** [reply]
In the cloud version, the values for these environment variables are already configured by the Chatwoot Team. It is mandatory to have separate configuration values for each self-hosted installation for the facebook inbox to work.

**Reply by ptd-31** [reply]
Hi @sojan-official , Thank you for answer. Can you suggest for me about configuration values for my self-hosted for facebook inbox to work with multi accounts Facebook? Because, if i change value old(FB_VERIFY_TOKEN, FB_APP_SECRET, FB_APP_ID) then old  fanpage in app Facebook disconect(can't send or receive message)...

**Reply by sojan-official** [reply]
Multiple pages can work with a single Facebook app.  But if you want to connect pages where you don't have admin access, you will need to get permissions approved from Facebook, as described above. 

Also, if you change the values for the environment variables described above, you will have to reconnect your inboxes

**Reply by ptd-31** [reply]
Thanks you, what permissions do i need facebook to approve to connect pages without admin access? @sojan-official

**Reply by tejaswinichile** [reply]
Hey @ptd-31 these are the permission that you would need to have the access to facebook pages if you don't have admin permission.
`pages_show_list, pages_manage_metadata, pages_messaging, Business Asset User Profile Access`

**Reply by ptd-31** [reply]
Thanks you so much for reply @tejaswinichile .
Facebook approved my app: pages_show_list, pages_manage_metadata, pages_messaging, Business Asset User Profile Access(as photo). But when another users(they don't have role in the my app Facebook.) can't show list page when they create inbox messenger? If they are in the role then can show list page. How without adding users to the role they can still show their list page to create messenger inbox? I hope to receive from you soon. Thanks
![fb](https://user-images.githubusercontent.com/57672922/193441384-259747fd-18f4-4fea-a737-e59d36db07b5.png)

**Reply by tejaswinichile** [reply]
I didn't get your question completely but here is how it works, So when a user login with Facebook while adding a Facebook page inbox in the Chatwoot dashboard they give permission to the app to access the pages and that's how they can add their own pages to the app.

**Reply by ouzbalkn** [reply]
> it works, So when a user login with Facebook while adding a Facebook page inbox in the Chatwoot dashboard they give permission to the app to access the pages and that's how they can add their own pages to the app.

How did you get all these access?

**Comment by luisdanielhj** [comment]
I am receiving the messages that are made directly to the page, but when a message is made through an advertisement on the page, I do not receive it

{Correction} i am just receiving messages if i the creator of the facebook app send a message to the page, if other profile do this the crm dont receive the message

**Reply by luiszxcv** [reply]
Como resolver?

**Reply by luiszxcv** [reply]
> Estou recebendo as mensagens que são feitas diretamente na página, mas quando uma mensagem é feita através de um anúncio na página, não recebo
> 
> {Correção} só estou recebendo mensagens se eu o criador do app do facebook mandar uma mensagem para a página, se outro perfil fizer isso o crm não recebe a mensagem

Mesmo problema aqui.

**Reply by laerciogentil** [reply]
Did you manage to solve the problem?

**Comment by pandigita** [comment]
I'm trying to configure chatwoot (self hosted, of course) to integrate with both Facebook (Messenger) and Instagram. The docs foresee a very similar process, but my understanding is I still create two apps for the two inside of the Meta Developer dashboard, and that I have to do the process twice.
However, the docs say I should add `FB_APP_SECRET=` and `FB_APP_ID=` for Facebook, but it also says I should add exactly the same constants for Instagram! Is that a spelling mistake, and it should be something like `IG_APP_SECRET=` ? IG instead of FB?
Obviously I can't declare the same variable twice, right? So what's going on here exactly? 

Any help highly appreciated! And sorry if this is a silly oversight on my end.

**Comment by shivamkb17** [comment]
I have approved app of facebook but Instagram message feature getting disapproved, Reason is improproper handling of CDN resources. 
**App rejected - Improper handling of media CDN URLs**

Tag user show in URL formate, like; "
?asset_id=17958331058424642&signature=AbzKFbAMdrLpSsPXSUG7IetfrDFtxjAAzds4y0dSxTBNHt7Yh0ivV305vurX6x_zNZYF9ijwXQ36vN6ono7-xuMwMeMSXGWe9km_MtBCD2k0aFkp6dGwh3vDUnzTFKjBIQpueMFW5Luof5IVmFDV0z8DpM4V1lJDU6yyMEedchgQ9OA8SurPlE5T-DurOxxxxxxxxxxxxxxxxxWXDIwP8oh3Fi "


instagram_manage_messages 
**feedback**

**App rejected - Improper handling of media CDN URLs**
[Developer Policy 8.9.c - Follow technical documentation](https://developers.facebook.com/devpolicy/)
During our review, we found that your bot does not comply with the quality guidelines and other specific requirements for Instagram Messaging in our technical documentation. Specifically, proper handling of media CDN URLs by not storing/caching the media content.
To resolve this issue, update your app to follow our technical documentation, then re-submit for review.
For more details on submission requirements please read our [Messenger Platform documentation](https://developers.facebook.com/docs/messenger-platform/overview) and our app review [documentation for Instagram Messaging](https://developers.facebook.com/docs/messenger-platform/instagram/app-review).

**Reply by tejaswinichile** [reply]
@shivamkb17 The image takes time to load because we'll fetch the story on the fly, according to Facebook's suggestion. We don't store their CDN URL and delete the same when the story is invalid.

**Reply by danilosong** [reply]
Did you manage to solve it? I have the same problem

**Comment by growsmartmartech** [comment]
@tejaswinichile I'm going through the same difficulties implementing Facebook and Instagram integration. I've followed the community tips and still haven't made progress. You made a video demonstrating a part of the process, could you do it for the whole process? I believe it would help everyone who goes through this difficulty. Probably because I don't know the right path.

Thanks!

**Reply by Sixth2538** [reply]
@tejaswinichile Would be great if we can have an updated guide once Chatwoot V3 is pushed! Having trouble with facebook/instagram advanced app permission as well.

**Reply by tejaswinichile** [reply]
What is the issue that the Facebook team said about the rejection?

**Comment by zhiru** [comment]
I'm having problems to test facebook messages, instagram is working but not facebook

**Reply by tejaswinichile** [reply]
Do you see webhooks reaching to the server?

**Comment by mv-developer** [comment]
@tejaswinichile 
I ENCOUNTERED AN ERROR when I attempted to request permissions and submit my app for review. I tried to reach out to Facebook support for assistance, but unfortunately, I couldn't find any contact information related to that particular issue.

<img width="802" alt="Screenshot 2023-07-09 at 9 22 21 PM" src="https://github.com/chatwoot/chatwoot/assets/2767707/a97ba221-6f91-41f6-81e9-a67db56e4727">
<img width="682" alt="Screenshot 2023-07-09 at 9 22 57 PM" src="https://github.com/chatwoot/chatwoot/assets/2767707/38de8824-78ba-490e-b8fb-89fcea5d3100">

**Comment by rfratucci** [comment]
Hi!
I've tried about ten attempts to approval of advanced permissions, without sucess. Anyone knows any developer or consulting thats may do this process for me?

**Comment by ycarooliveira** [comment]
I am having problem with instagram permissions as well, they are saying that I dont need the permissions I'm asking, that are:

_pages_show_list
pages_manage_metadata
pages_messaging
instagram_manage_messages
pages_read_engagement
instagram_basic_

**Comment by VonJdamian** [comment]
How can I solve this CDN problem?

![image](https://github.com/chatwoot/chatwoot/assets/142674805/b538494a-d7e3-4332-8465-e0af9c0b113b)

**Reply by pelucapampa** [reply]
Anyone have solution for that?

I'm having problems to approval process for following permission:
Business Asset User Profile Access 
pages_manage_metadata 
pages_read_engagement
instagram_manage_messages 
instagram_basic

I sent the instructions follow this post, but all request are rejected by facebook team.

Have you other guide to make facebook app approved?

**Comment by lazmo88** [comment]
Hello, just to clarify something. If I am the admin of the pages I want to add, and I am also developer of the FB app do I need to go through the verification process of the app to get it working? Reason why I am asking, I followed all the steps, and added credentials to CW env, FB login works, it asks for permissions, and lets me select pages/IG profiles.

Problem is the pages list is empty on CW dashboard. None of the pages/profiles are listed and I am unable to complete the integration. pages_show_list permission is active and on standard level. On Meta developer portal it states "Your app only needs advanced access to the following permissions and features if your business is a Tech Provider. " - if I understand this correctly, advanced access would be required only if I want to offer the app to 3rd parties?

Is this empty list a known bug as it is reported few times: #7491 and #3100 

@sojan-official @tejaswinichile your input would be highly appreciated :)

Edit: So, I tried creating test app again and page list is now working. Only issue I have is that incoming Instagram messages are duplicated. So incoming message is displayed twice on CW, replies are sent only once. Is it still necessary to have separate IG messaging webhooks active if the FB page and IG profile are already connected?

**Reply by lazmo88** [reply]
So, disabling Instagram specific webhooks solved the duplicate incoming messages issue. What is the reason docs guide you to use Instagram webhooks?

**Reply by ailsoncgt** [reply]
I am facing the same problem with instagram message. The workaround was caching sender and time to each webhook event :/

**Comment by pelucapampa** [comment]
Check the new Facebook Pages Permissions.

Yo need to invite to manage your page from this

Go to you Page on Facebook
Setup and Privacy
New Experience for Pages

Here you need to add a user for Facebook Access
________________________________
De: lazmo88 ***@***.***>
Enviado: jueves, 23 de noviembre de 2023 06:43
Para: chatwoot/chatwoot ***@***.***>
Cc: pelucapampa ***@***.***>; Comment ***@***.***>
Asunto: Re: [chatwoot/chatwoot] Issues with Messenger Integration (Discussion #3397)


Hello, just to clarify something. If I am the admin of the pages I want to add, and I am also developer of the FB app do I need to go through the verification process of the app to get it working? Reason why I am asking, I followed all the steps, and added credentials to CW env, FB login works, it asks for permissions, and lets me select pages/IG profiles.

Problem is the pages list is empty on CW dashboard. None of the pages/profiles are listed and I am unable to complete the integration. pages_show_list permission is active and on standard level. On Meta developer portal it states "Your app only needs advanced access to the following permissions and features if your business is a Tech Provider. " - if I understand this correctly, advanced access would be required only if I want to offer the app to 3rd parties?

Is this empty list a known bug as it is reported few times: #7491<https://github.com/orgs/chatwoot/discussions/7491> and #3100<https://github.com/orgs/chatwoot/discussions/3100>

@sojan-official<https://github.com/sojan-official> @tejaswinichile<https://github.com/tejaswinichile> your input would be highly appreciated :)

**Comment by salman652** [comment]
Hi @tejaswinichile,  i hope you are doing well. i'm getting issue to approve "business_management" permission.

FB raise issue on purpose for this permission. Can you please guide me what are the suitable use case for approve this permission. And it will be helpful if you provide me submission details for "business_management" and screen-cast too.
Thank You!
![busniess_management](https://github.com/chatwoot/chatwoot/assets/40527000/c3252473-df3b-495f-8c2a-7675e2aa63da)

**Reply by pelucapampa** [reply]
business_management

Dinos cómo usas este permiso o función
We use this permission to fetch pages managed by a business along with pages show list

To test that, please login on [https://yourchatwootinstance.com/]
Use following credentials:
User: xxx@xxxx
Password: PassXXX

1) Then, you need to add inbox, and here we need permission, when user add a new Inbox, our app need fetch page managed.

2) Go to Inbox section and you can see "+ New inbox" click here and then select "Messenger" to add a Facebook or instagram Pages

3) Click on "Continue with Facebook" and Insert Facebook credentials. (On past review submit you cannot use this credentials, but are correct, please try again)
In this case, PLEASE use this Facebook Test account credentials
User: userxxx
Pass: PassXXX

4) When you are logged, our app show a List of pages to select one of them.

5) Select one of the Page's, and click Next

6) Select agent, and click Next

**Comment by tobsowo** [comment]
@tejaswinichile please what would you advise to respond with for `instagram_manage_messages` asking `How an app handles when user unsends messages`?

**Reply by tejaswinichile** [reply]
https://developers.facebook.com/docs/messenger-platform/instagram/features/webhook With the latest update you can check messaging webhook with is_deleted flag.

**Reply by tobsowo** [reply]
I tried to delete a message sent from IG but it wasn't on Chatwoot.

**Reply by bielvelozo** [reply]
The same thing happened to me, I'm using version 3.8.0

**Reply by stephanto-dev** [reply]
same for me

**Reply by ailsoncgt** [reply]
I think message deleted webhook just work on mobile app.

**Reply by lafi87** [reply]
Any news about this? In the review request it requires showing the deletion process but the messages cannot be deleted.

**Reply by tobsowo** [reply]
It works on IG mobile app, so use that for your demo video

**Comment by md-riaz** [comment]
pages_show_list feedback
Notes from your reviewer:
Hello Team,


We have determined your use case is valid. While reviewing your submission, we were unable to verify use case in your screencast. Please provide screencast where your use case is reflected. Please show pages

Please re-submit accordingly and we will re-review your application. Thanks!

How to solve it? currently non admin users cannot add their facebook pages and new messages are coming as John Doe names. Anyone knows why?

**Comment by md-riaz** [comment]
pages_read_engagement feedback
Notes from your reviewer:
Hi Team,

We were unable to approved this permission due to the use case mentioned is unclear/invalid. It would be preferable if you provide a detailed explanation of how you intend to use this permission in your app.

This permission allows your app to read content (posts, photos, videos, events) posted by the Page, read followers data, and profile picture, and read metadata and other insights about the Page. Display Facebook post to website and retrieve page access token.

*Allowed Usage*
- Get content posted by your Page.
- Get names, PSIDs, and profile pictures of your Page followers.
- Get metadata about your Page.

Here’s a resource that may help:
https://developers.facebook.com/docs/permissions/reference/pages_read_engagement

Please re-submit accordingly and we will re-review your application. Thank you!

**Comment by tobsowo** [comment]
instagram_manage_messages feedback

```Hi Team!

We're unable to approve your permission request even your usecase are valid because we're unable to replicate and fully verify the functions of the permission. **App failed to capture pages after integration.**

Please re-submit accordingly and we will re-review your application.

Thank you!
```

The above is the response from Meta, can someone please explain to me what **App failed to capture pages after integration**  means.

**Comment by Braingearceo** [comment]
We are having problem with Meta submission for business_management approval. It's pain in the ass to work with Meta """"""" Reviewer """""". Never giving any explanation, always returning a refusal that as even nothing to do with their own comment. I have trace of them saying that they tried but our credentials was not working, even if the server logs show they did not even bothered to ping the page.

They accepted all permission but Business_management. So we have access to advanced stuff that we cannot use because business_management permission is mandatory to fetch pages data such as access_token and id (otherwise me/accounts return an empty array since v18.0 of the graph api.) And they keep saying that this permission is not required for our application. Like the .........??

So far we have add subtitle in the video uploaded at the moment we use the permission, we are describing why we use this permission, including the actual problem with the API endpoint that should return data but do not due to missing business_management permission. 

Yet we can't contact anyone. Anyone have faced the same problem? Like we got accepted for 6 other permission but not this one. Are reviewer living in another timeline where 18 of the endpoint doesn't require advanced access?
<img width="963" alt="Capture d’écran 2024-06-12 à 10 01 29 - copie" src="https://github.com/chatwoot/chatwoot/assets/104493450/6f1dd7c8-b908-47fe-9c97-b6a8dc98ec6d">
<img width="961" alt="Capture d’écran 2024-06-12 à 10 04 46" src="https://github.com/chatwoot/chatwoot/assets/104493450/1e598782-27bb-4bad-94cb-8ea391e88960">


From valid to not valid in less than 24hours. From yeah it's fine you just need to convert the instruction to French. To yeah it's fine, you just need to put the instruction in the other slot. 

Come on, even had to use a vpn to philippine to allow them to connect to a test account that we need to break meta rule to create in order to allow meta reviewer to test our application. It's a complete joke.

**Comment by KgySathsara** [comment]
How will this app use pages_show_list?

How will this app use pages_manage_metadata?

How will this app use pages_utility_messaging?

How will this app use pages_messaging?

How will this app use business_management?

How will this app use pages_read_engagement?


please help me , how to send this short video
